src/main/java/controllers/CommandLineInterface.java

```diff
@@ -18,6 +18,37 @@
 public class CommandLineInterface {
 
     public CommandLineInterface() {}
+    private int generationMode;
```

Comment:
Think this would be better with an enum, so you can explicitly name the choices (so like ONE_BY_ONE and ALL_PERMUTATIONS) or something. So we don't have to mentally map 0 and 1 to all permutations and one by one.

Reply:
if u want an example of what I'm thinking, basically the Day enum from the TimeFilter

Reply:
also i think u can give it a `toString`, so if u ever need to print to CLI what generation mode you're in, don't need to convert.

Reply:
Very nice idea thank you